src/server/cmd/pachctl/cmd/cmd.go

```diff
@@ -180,9 +188,135 @@ kubectl %v port-forward "$pod" %d:8081
+
+The migration command takes the general form:
+
+$ pachctl migrate --from <FROM_VERSION> --to <TO_VERSION>
+
+If "--from" is not provided, pachctl will attempt to discover the current
+version of the cluster.  If "--to" is not provided, pachctl will use the
+version of pachctl itself.
+
+Example:
+
+# Migrate Pachyderm from 1.4.8 to 1.5.0
+$ pachctl migrate --from 1.4.8 --to 1.5.0
+`,
+		Run: cmdutil.RunFixedArgs(0, func(args []string) (retErr error) {
+			// If `from` is not provided, we use the cluster version.
+			if from == "" {
+				versionClient, err := getVersionAPIClient(address)
+				if err != nil {
+					return sanitizeErr(err)
+				}
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+				defer cancel()
+				clusterVersion, err := versionClient.GetVersion(ctx, &types.Empty{})
+				if err != nil {
+					return fmt.Errorf("Unable to discover cluster version; please provide the --from flag.  Error: %v\n", err)
+				}
+				from = version.PrettyPrintVersionNoAdditional(clusterVersion)
+			}
+
+			// if `to` is not provided, we use the version of pachctl itself.
+			if to == "" {
+				to = version.PrettyPrintVersionNoAdditional(version.Version)
+			}
+
+			jobSpec := batch.Job{
+				TypeMeta: unversioned.TypeMeta{
+					Kind:       "Job",
+					APIVersion: "batch/v1",
+				},
+				ObjectMeta: api.ObjectMeta{
+					Name: "pach-migration",
+					Labels: map[string]string{
+						"suite": "pachyderm",
+					},
+				},
+				Spec: batch.JobSpec{
+					Template: api.PodTemplateSpec{
+						Spec: api.PodSpec{
+							Containers: []api.Container{
+								{
+									Name:    "migration",
+									Image:   fmt.Sprintf("pachyderm/pachd:%v", version.PrettyPrintVersion(version.Version)),
+									Command: []string{"/pachd", fmt.Sprintf("--migrate=%v-%v", from, to)},
+								},
+							},
+							RestartPolicy: "OnFailure",
+						},
+					},
+				},
+			}
+
+			tmpFile, err := ioutil.TempFile("", "")
+			if err != nil {
+				return err
+			}
+			defer os.Remove(tmpFile.Name())
+
+			jsonEncoderHandle := &codec.JsonHandle{
+				BasicHandle: codec.BasicHandle{
+					EncodeOptions: codec.EncodeOptions{Canonical: true},
+				},
+				Indent: 2,
+			}
+			encoder := codec.NewEncoder(tmpFile, jsonEncoderHandle)
+			jobSpec.CodecEncodeSelf(encoder)
+			tmpFile.Close()
+
+			cmd := exec.Command("kubectl", "create", "--validate=false", "-f", tmpFile.Name())
+			out, err := cmd.CombinedOutput()
+			fmt.Println(string(out))
+			if err != nil {
+				return err
+			}
+			fmt.Println("Successfully launched migration.  To see the progress, use `kubectl logs job/pach-migration`")
+			return nil
+		}),
+	}
+	migrate.Flags().StringVar(&from, "from", "", "The current version of the cluster.  If not specified, pachctl will attempt to discover the version of the cluster.")
+	migrate.Flags().StringVar(&to, "to", "", "The version of Pachyderm to migrate to.  If not specified, pachctl will use its own version.")
```

Comment:
I think they could be useful.  We need the `from` flag because someone might be running the migration after they've redeployed their cluster, in which case we need the user to tell us what version they were originally using.  Hiding `to` from the user also seems unnecessary since we do have the ability to launch migration between any two versions, and if the user doesn't specify `to` it just uses pachctl's own version.  I don't think we need to intentionally hide power from the user.